### PR TITLE
don't run delete statement if array is empty

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team @macbre 
https://wikia-inc.atlassian.net/browse/SERVICES-535

I suspect this db query was throwing an error when the preferences list was empty

### CODE REVIEW

File: includes/User.php [hunk 1]
```diff
@@ -4851,11 +4851,15 @@ protected function saveOptions() {
...
-				->WHERE('up_user')->EQUAL_TO($this->getId())
-					->AND_('up_property')->NOT_IN($preferenceNames)
-				->run($dbw);
+				->WHERE('up_user')->EQUAL_TO($this->getId());
+
+			if (!empty($preferenceNames)) {
+				$sql->AND_('up_property')->NOT_IN($preferenceNames);
+			}
+
+			$sql->run($dbw);
```
Comment (drsnyder) [inline includes/User.php:4862]: Seems like this is also a bug with `WikiaSQL`. Should an `IN` operation be a noop if the scalar passed is empty?

**Comment by drsnyder** [comment]
LGTM

File: includes/User.php (hunk 1 above)
Comment (nmonterroso) [inline includes/User.php:4862]: Maybe, but I'd rather it throw an error and alert (like it did), instead of silently throwing queries on the ground.

Comment (drsnyder) [inline includes/User.php:4862]: Let the queries hit the floor!
let the queries hit the floor

Comment (macbre) [inline includes/User.php:4862]: Making a DELETE query in `saveOptions` method is a bit counter-intuitive :) Can we add a comment here describing why we're doing that?

File: includes/User.php [hunk 2]
```diff
@@ -4851,11 +4851,15 @@ protected function saveOptions() {
 		if ($wgPreferencesUseService) {
```
Comment (macbre) [inline includes/User.php:4851]: Please consider using `UserSaveOptions` hook instead (or introducing a new one).

File: includes/User.php (hunk 1 above)
Comment (nmonterroso) [inline includes/User.php:4862]: I agree, it's definitely not something you'd expect to happen. Though, we only added it to maintain compatibility with what MediaWiki was doing (see the `else` statement)

**Pull request merged**